code/game/objects/items/teleportation.dm

```diff
@@ -312,3 +312,154 @@
+		to_chat(user, "<span class='notice'>\The [src] is malfunctioning.</span>")
+		return
+
+	var/mob/living/carbon/C = user
+	var/teleport_distance = rand(minimum_teleport_distance,maximum_teleport_distance)
+	var/turf/destination = get_teleport_loc(current_location,C,teleport_distance,0,0,0,0,0,0)
+	var/list/bagholding = user.GetAllContents(/obj/item/storage/backpack/holding)
+
+	if(isclosedturf(destination))
+		if(!EMP_D && !(bagholding.len))
+			panic_teleport(user, destination) //We're in a wall, engage emergency parallel teleport.
+		else
+			get_fragged(user, destination) //EMP teleported you into a wall? Wearing a BoH? You're dead.
+	else
+		telefrag(destination, user)
+		do_teleport(C, destination, channel = TELEPORT_CHANNEL_FREE)
+		charges--
+		check_charges()
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(current_location)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+		playsound(destination, 'sound/effects/phasein.ogg', 25, 1)
+		playsound(destination, "sparks", 50, 1)
+
+/obj/item/teleporter/proc/panic_teleport(mob/user, turf/destination)
+	var/mob/living/carbon/C = user
+	var/turf/mobloc = get_turf(C)
+	var/turf/emergency_destination = get_teleport_loc(destination,C,0,0,1,saving_throw_distance,0,0,0)
+
+	if(emergency_destination)
+		telefrag(emergency_destination, user)
+		do_teleport(C, emergency_destination, channel = TELEPORT_CHANNEL_FREE)
+		charges--
+		check_charges()
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+		new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(emergency_destination)
+		playsound(emergency_destination, 'sound/effects/phasein.ogg', 25, 1)
+		playsound(emergency_destination, "sparks", 50, 1)
+	else //We tried to save. We failed. Death time.
+		get_fragged(user, destination)
+
+/obj/item/teleporter/proc/get_fragged(mob/user, turf/destination)
+	var/turf/mobloc = get_turf(user)
+	user.forceMove(destination)
+	playsound(mobloc, "sparks", 50, TRUE)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(mobloc)
+	new /obj/effect/temp_visual/teleport_abductor/syndi_teleporter(destination)
+	playsound(destination, "sparks", 50, TRUE)
+	playsound(destination, "sound/magic/disintegrate.ogg", 50, TRUE)
+	to_chat(user, "<span class='userdanger'>You teleport into the wall, the teleporter tries to save you, but-</span>")
+	destination.ex_act(2) //Destroy the wall
+	if (iscarbon(user))
+		var/mob/living/carbon/cuser = user
+		cuser.gib(FALSE,FALSE,FALSE,FALSE)
+
+/obj/item/teleporter/proc/telefrag(turf/fragging_location, mob/user)
+	for(var/mob/living/M in fragging_location)//Hit everything in the turf
+		M.apply_damage(20, BRUTE)
+		M.Paralyze(30)
+		to_chat(M, "<span class='userdanger'>[user] teleports into you, knocking you to the floor with the bluespace wave!</span>")
```

Comment:
`to_chat(M, span_userdanger("[user] teleports into you, knocking you to the floor with the bluespace wave!"))`